portafly/src/components/util/AlertsContext.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useContext } from 'react'
+import {
+  AlertGroup,
+  AlertActionCloseButton,
+  Alert,
+  AlertVariant
+} from '@patternfly/react-core'
+
+interface IAlert {
+  id: string
+  variant: keyof typeof AlertVariant
+  title: string
+  timeout: NodeJS.Timeout
+}
+
+type IAlertsContext = { addAlert: (alert: Omit<IAlert, 'timeout'>) => void }
```

Comment:
I feel like when using `Omit` to remove some properties, and it's not for defining a new type or interface, I'm missing the "why" or "what"... specially when it's used more than once. Maybe declaring an `Alert` type, and extend it in the interface? Utility types are a double edge sword sometimes

Reply:
Well the why is because it doesn't make sense to define 2 types for Alerts. `timeout` is only relevant for the Provider so as you mention the `Alert` shouldn't have this. But I didn't do it at first because it turned out to be very verbose:
```ts
React.useState<(IAlert & { timeout: NodeJS.Timeout})[]>([])
```